sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala

```diff
@@ -74,3 +84,8 @@ case class ResolvedTable(catalog: TableCatalog, identifier: Identifier, table: T
 case class ResolvedView(identifier: Identifier) extends LeafNode {
   override def output: Seq[Attribute] = Nil
 }
+
+case class ResolvedFunc(catalog: CatalogPlugin, functionIdentifier: FunctionIdentifier)
```

Comment:
shall we put `CatalogFunction` instead of `FunctionIdentifier` in the parameter? otherwise the lookup is still done at runtime.

Reply:
Added it.

BTW do we need a `FunctionCatalog` like `TableCatalog` in v2 ?